alf/algorithms/algorithm.py

```diff
@@ -1255,17 +1255,22 @@ def _train_experience(self, experience, batch_info, num_updates,
             common.warning_once(
                 "Experience length has been cut to %s" % length)
 
-        if len(alf.nest.flatten(
-                self.train_state_spec)) > 0 and not self._use_rollout_state:
-            if mini_batch_length == 1:
-                logging.fatal(
-                    "Should use TrainerConfig.use_rollout_state=True "
-                    "for training from a replay buffer when minibatch_length==1, "
-                    "otherwise the initial states are always zeros!")
-            else:
+        if len(alf.nest.flatten(self.train_state_spec)) > 0:
```

Comment:
It would be useful to explicitly list proper setting for some common examples case here, e.g.  1) with/without RNN 2) mini_batch_length ==1 / > 1, maybe as a table.

Reply:
added comments